diff --git a/dist/compiler/server/write.js b/dist/compiler/server/write.js
index 625502ab5e66aa55b8c67795aa9d0baa773ed3ea..8c2a3359257432b863fdd68d94db081e6058105f 100644
--- a/dist/compiler/server/write.js
+++ b/dist/compiler/server/write.js
@@ -44,7 +44,7 @@ async function write(config, outputFiles) {
     if ([".js", ".cjs", ".mjs"].some(ext => file.path.endsWith(ext))) {
       // fix sourceMappingURL to be relative to current path instead of /build
       let filename = file.path.substring(file.path.lastIndexOf(path__namespace.sep) + 1);
-      let escapedFilename = filename.replace(/\./g, "\\.");
+      let escapedFilename = filename.replace(/([.\[\]])/g, "\\$1");
       let pattern = `(//# sourceMappingURL=)(.*)${escapedFilename}`;
       let contents = Buffer.from(file.contents).toString("utf-8");
       contents = contents.replace(new RegExp(pattern), `$1${filename}`);
